embedded-graphics/src/primitives/triangle/mod.rs

```diff
@@ -88,9 +88,48 @@ impl ContainsPoint for Triangle {
             return false;
         }
 
-        // This is inefficient and should be replaced by a better algorithm to
-        // determine if point is inside the triangle
-        self.points().any(|p| p == point)
+        let p = point;
+        let Self { p1, p2, p3 } = *self;
+
+        // Check if point is inside triangle using https://stackoverflow.com/a/20861130/383609.
+        // Works for any point ordering.
+        let is_inside = {
+            let s = p1.y * p3.x - p1.x * p3.y + (p3.y - p1.y) * p.x + (p1.x - p3.x) * p.y;
+            let t = p1.x * p2.y - p1.y * p2.x + (p1.y - p2.y) * p.x + (p2.x - p1.x) * p.y;
+
+            if (s < 0) != (t < 0) {
```

Comment:
Just a fun thing, this is compiled to exactly the same machine code as `same_signs` in the line joiners PR.

https://godbolt.org/z/EMGGEM

Reply:
Cool! I'll likely factor this logic out in that PR as they're doing the same thing.